ca/ca.go

```diff
@@ -104,15 +105,11 @@ type certificateAuthorityImpl struct {
 // that, if two issuers have the same nearly-unique ID, the *latter* one in
 // the input list "wins".
 func makeIssuerMaps(issuers []*issuance.Issuer) issuerMaps {
-	issuersByAlg := make(map[x509.PublicKeyAlgorithm]*issuance.Issuer, 2)
+	issuersByAlg := make(map[x509.PublicKeyAlgorithm][]*issuance.Issuer, 2)
 	issuersByNameID := make(map[issuance.NameID]*issuance.Issuer, len(issuers))
 	for _, issuer := range issuers {
-		for _, alg := range issuer.Algs() {
-			// TODO(#5259): Enforce that there is only one issuer for each algorithm,
-			// instead of taking the first issuer for each algorithm type.
-			if issuersByAlg[alg] == nil {
-				issuersByAlg[alg] = issuer
-			}
+		if issuer.IsActive() {
+			issuersByAlg[issuer.KeyType()] = append(issuersByAlg[issuer.KeyType()], issuer)
```

Comment:
Should duplicate values in `[]issuance.IssuerConfig` be accepted? In the [multiple certificate profiles PR](https://github.com/letsencrypt/boulder/pull/7325/files#diff-9033031dd64a1f3c17bc6f97c5e0c3a0375eafa6e81bbbbbaccedc6446d94c8dR168-R177) we bailed out if there was a duplicate entry. If you decide to do this, the function comment should also be changed.